components/movementsensor/gpsnmea/gpsnmea.go

```diff
@@ -81,16 +76,12 @@ func (cfg *Config) Validate(path string) ([]string, error) {
 
 // ValidateI2C ensures all parts of the config are valid.
 func (cfg *I2CConfig) validateI2C(path string) error {
-	if cfg.I2CBus == "" {
+	if cfg.I2CBus == 0 {
```

Comment:
i2c bus 0 is a valid i2c bus on many boards, and is frequently the default

Reply:
Oh dang, great catch! I'll update this so the config stores a pointer, and we check if it's nil or not.

Reply:
This should just be a string. There's nothing enforcing that i2c buses are numbered; everything all the way down to `i2creg.Open` takes a string as the devicename. https://pkg.go.dev/periph.io/x/conn/v3/i2c/i2creg

Note that `genericlinux.GetI2CBus` is also broken because it forbids bus 0.

Reply:
Also, if not 0, all that `genericlinux.GetI2CBus` does with this number is convert it directly back to a string.

Reply:
You make a fair point that Periph's `i2creg` library eventually takes a string, and it's intuitive to keep the same type all the way up. I'm not convinced that it should ever be a non-numerical value (Periph merely gives you the option of specifying `"7"` or `"/dev/i2c-7"`, and why they give you that option is beyond me), but there is merit in consistency. I'll change it back to a string...

Reply:
Linux being linux, I'm almost certain there's a way that a user could also change the name of their i2c devices to be `/dev/i2c-name` especially with a lot of the custom overlay stuff a lot of SBCs do.

Reply:
Ah. and with that change, I need to first go make a PR to update the PCA9685 board, which already has had this treatment applied, and tweak the scope doc to match. Thanks for bringing this up! We totally missed it in the past.